pkg/controller/cronworkflow.go

```diff
@@ -0,0 +1,306 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/golang/glog"
+
+	apiv1 "k8s.io/api/core/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	v1core "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/workqueue"
+
+	cwapi "github.com/amadeusitgroup/workflow-controller/pkg/api/cronworkflow/v1"
+	wapi "github.com/amadeusitgroup/workflow-controller/pkg/api/workflow/v1"
+
+	wclientset "github.com/amadeusitgroup/workflow-controller/pkg/client/clientset/versioned"
+	cwlisters "github.com/amadeusitgroup/workflow-controller/pkg/client/listers/cronworkflow/v1"
+	wlisters "github.com/amadeusitgroup/workflow-controller/pkg/client/listers/workflow/v1"
+)
+
+// CronWorkflowControllerConfig contains info to customize Workflow controller behaviour
+type CronWorkflowControllerConfig struct {
+	RemoveIvalidCronWorkflow bool
+	NumberOfThreads          int
+}
+
+// CronWorkflowController represents the Workflow controller
+type CronWorkflowController struct {
+	CronWorkflowClient wclientset.Interface
+	KubeClient         clientset.Interface
+	queue              workqueue.RateLimitingInterface // Workflows to be synced
+
+	CronWorkflowLister cwlisters.CronWorkflowLister
+	CronWorkflowSynced cache.InformerSynced
+
+	WorkflowLister wlisters.WorkflowLister
+	WorkflowSynced cache.InformerSynced // returns true if job has been synced. Added as member for testing
+
+	updateHandler func(*cwapi.CronWorkflow) error // callback to upate Workflow. Added as member for testing
+	syncHandler   func(string) error
+
+	Recorder record.EventRecorder
+	config   CronWorkflowControllerConfig
+}
+
+// NewCronWorkflowController creates and initializes the CronWorkflowController instance
+func NewCronWorkflowController(
```

Comment:
I think this function should take a CronWorkflowControllerConfig as a parameter

Reply:
right, but WorkflowControlleris the same. Going to do in another PR for both